Gems/Maestro/Code/Source/Cinematics/AnimSequence.cpp

```diff
@@ -387,13 +387,14 @@ void CAnimSequence::RemoveNode(IAnimNode* node, bool removeChildRelationships)
     {
         if (node == m_nodes[i].get())
         {
-            m_nodes.erase(m_nodes.begin() + i);
-
             if (node->NeedToRender())
             {
                 RemoveNodeNeedToRender(node);
             }
 
+            // TODO : move destruction after usage in CUiAnimSequence::RemoveNode() too ?
```

Comment:
Put the comment rather in CUiAnimSequence::RemoveNode() and use the intial issue such as 
// TODO #11500 Moving node destruction after usage below might help to fix sequence entity spawning, but needs more investigation

Reply:
Thanks, this surely makes great sense.
Moved the TODO comment to the CUiAnimSequence::RemoveNode() in commit  https://github.com/o3de/o3de/pull/18538/commits/c332944c92542710caac1015263422121fe5dc0e.